internal/fs/handle/file.go

```diff
@@ -47,16 +47,27 @@ type FileHandle struct {
 	// will be downloaded for random reads as well too.
 	cacheFileForRangeRead bool
 	metricHandle          common.MetricHandle
+	// For now, we will consider the files which are open in append mode also as write,
+	// as we are not doing anything special for append. When required we will
+	// define an enum instead of boolean to hold the type of open.
+	readOnly bool
 }
 
-func NewFileHandle(inode *inode.FileInode, fileCacheHandler *file.CacheHandler, cacheFileForRangeRead bool, metricHandle common.MetricHandle) (fh *FileHandle) {
+func NewFileHandle(
```

Comment:
Nit: Let's add a comment that this assumes inode lock. Same for Destroy. `LOCKS_REQUIRED(fh.inode)`

Also method declaration should be on a single line to avoid indentation confusion https://google.github.io/styleguide/go/decisions#function-formatting